vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -109,6 +135,32 @@ def parse_pod(pod)
       new_result
     end
 
+    def parse_endpoint(entity)
+      new_result = parse_base_item(entity)
+      new_result[:container_groups] = []
+
+      # remove once kubernetes api v1beta3/v1.0 will be stable, see:
+      # https://github.com/GoogleCloudPlatform/kubernetes/commit/8ae20382
+      if !entity.respond_to?(:subsets) || entity.subsets.nil?
+        $log.warn("your kubernetes endpoints are not providing subsets, " \
+                  "please upgrade (see commit 8ae20382")
+        return new_result
+      end
+
+      entity.subsets.each do |subset|
+        subset.fetch('addresses', []).each do |address|
+          target_ref = address.fetch('targetRef', nil)
```

Comment:
fetch will nil is not necessary, and just makes the code harder to read.  This can just be `target_ref = address['targetRef']`